src/xunit.runner.visualstudio/Utility/RunSettings.cs

```diff
@@ -113,7 +113,8 @@ public bool IsMatchingTargetFramework()
 
 #if NETCOREAPP
             return string.IsNullOrWhiteSpace(TargetFrameworkVersion) ||// Short circuit on null since we don't have anything to detect, return true
-                   (TargetFrameworkVersion.StartsWith(".NETCoreApp,", StringComparison.OrdinalIgnoreCase) ||
+                   (TargetFrameworkVersion.StartsWith("net5", StringComparison.OrdinalIgnoreCase) ||
```

Comment:
What about net6? We will soon branch to it.

Reply:
You should have a follow up to fix this properly.

Reply:
@nkolev92 what is the correct fix for this?

Reply:
Copying my comment from the internal e-mail thread: 

```
I think the guidance would be to never depend on the short name, or anything NuGet generated, but depend on the TFI/TFV.
If you ever have the NuGetFramework type, use the `Framework` property.
```

Let's not the same mistake as NuGet that caused: https://github.com/NuGet/Home/issues/5154

Reply:
Though to be fair, this value comes from the MSTest platform and is not the TFM the user uses. It is the evaluated MSBuild property.

Reply:
yeah, I'm not 100% familiar with the code here, but the general rule is to use props like TFI etc. 

We probably have other places in the stack where this is being done, so we should do our best to weed them out.

Reply:
Then I need to fix this faster, or add another condition to this if, and to the two ifs that are in vstest that match runtime providers.